### ISSUE TITLE
[Appreciation and feature request] This is gold!

### ISSUE THREAD

**Issue by extraymond** [tommyod/streprogen#1]
Thanks for creating this wonderful package with nice documentation. Python and exercise science in the same page? This is automatically the most wonderful thing I've seen today.

It would be nice to have some features to make this more applicable for athletes or trainers. I hope it's not too rude to comment on such.

---

## Repetition Maximum based output

It's very common for trainers to stick to their training goal while shift the loads depending on their physiques status. They might train under slightly lower weight with more repetitions to avoid the potential risk of injury while getting constant progress.

[Repetition Maximum and it's equations](https://en.wikipedia.org/wiki/One-repetition_maximum)

Another potential application is that trainers might train for different function goals. Whereas small volume high repetition for endurance and high volume low repetition for explosiveness and strength.

[Instruction from different association](http://www.stack.com/a/the-perfect-rep-ranges-for-size-strength-and-power)

## Training protocol based on adaptation phase shift.

It's known that individuals with little to no training experience will gain muscular strength from neuromuscular adaptation which would take approximately  8 weeks.

After this phase, one would have to increase muscular strength more directly from muscle growth.

[A great read about strength-endurance-continuum](https://www.strengthandconditioningresearch.com/perspectives/strength-endurance-continuum/)

---

Thanks for how modular and accessible your code are constructed, I think it's highly possible to implement the above ideas into this tool.

Although I'm not a python expert, if there's anything I can help, I would love to volunteer.

p.s: English is not my native tongue so forgive me for being unclear if that happens.

**Comment by tommyod** [comment]
Thanks for the compliments.

I believe most of these features are accessible in `streprogen` currently, althought it might not be completely obvious to how use them.

- The `progress_func` keyword argument to the `Program` class lets the user manipulate the mapping from repetitions to 1RM. The default is the `streprogen.modeling.reps_to_intensity` function, which maps `reps` to `97.5 -4.8 * (reps - 1)` by default. This can be customized by changing the slope and constant, or the user can specify their own such as those from [One-repetition maximum - Wikipedia](https://en.wikipedia.org/wiki/One-repetition_maximum). If you wish to implement the equations from wikipediate into the `streprogen.modeling` module I would like to see that.

- Volume is controlled by the `reps_per_exercise` and `intensity` keyword arguments to the `Program` class. They can also be functions of time, or can be overridden for specific exercises. See the documentation for the `[Program](http://streprogen.readthedocs.io/en/latest/classes.html#the-program-class)` class and the [`DynamicExercise`](http://streprogen.readthedocs.io/en/latest/classes.html#the-dynamicexercise-class) class.

In summary, most of the features you mention and available, although perhaps not obvious. If you wish to add equations from the Wikipedia page I would like to see that.
